plugins/outputs/azure_data_explorer/azure_data_explorer_test.go

```diff
@@ -193,6 +196,111 @@ func TestWrite(t *testing.T) {
+		metricsGrouping           string
+		tableNameToExpectedResult map[string]string
+		expectedWriteError        string
+		createTables              bool
+		ingestionType             string
+	}{
+		{
+			name:                      "Valid metric",
+			inputMetric:               mockMetrics,
+			createTables:              true,
+			metricsGrouping:           tablePerMetric,
+			tableNameToExpectedResult: expectedResultMap,
+		},
+		{
+			name:                      "Don't create tables'",
+			inputMetric:               mockMetrics,
+			createTables:              false,
+			metricsGrouping:           tablePerMetric,
+			tableNameToExpectedResult: expectedResultMap,
+		},
+		{
+			name:                      "SingleTable metric grouping type",
+			inputMetric:               mockMetrics,
+			createTables:              true,
+			metricsGrouping:           singleTable,
+			tableNameToExpectedResult: expectedResultMap,
+		},
+		{
+			name:                      "Valid metric managed ingestion",
+			inputMetric:               mockMetrics,
+			createTables:              true,
+			metricsGrouping:           tablePerMetric,
+			tableNameToExpectedResult: expectedResultMap,
+			ingestionType:             managedIngestion,
+		},
+		{
+			name:                      "Table per metric type",
+			inputMetric:               mockMetricsMulti,
+			createTables:              true,
+			metricsGrouping:           tablePerMetric,
+			tableNameToExpectedResult: expectedResultMap2,
+		},
+	}
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			//t.Parallel()
+			serializer, err := telegrafJson.NewSerializer(time.Second, "", "")
+			require.NoError(t, err)
+			for tableName, jsonValue := range testCase.tableNameToExpectedResult {
+				ingestionType := "queued"
+				if testCase.ingestionType != "" {
+					ingestionType = testCase.ingestionType
+				}
+				mockIngestor := &mockIngestor{}
+				plugin := AzureDataExplorer{
+					Endpoint:        "someendpoint",
+					Database:        "databasename",
+					Log:             testutil.Logger{},
+					IngestionType:   ingestionType,
+					MetricsGrouping: testCase.metricsGrouping,
+					TableName:       tableName,
+					CreateTables:    testCase.createTables,
+					kustoClient:     fakeClient,
+					metricIngestors: map[string]ingest.Ingestor{
+						tableName: mockIngestor,
+					},
+					serializer: serializer,
+				}
+				errorInWrite := plugin.Write(testCase.inputMetric)
+				if testCase.expectedWriteError != "" {
+					require.EqualError(t, errorInWrite, testCase.expectedWriteError)
+				} else {
+					require.NoError(t, errorInWrite)
+					createdIngestor := plugin.metricIngestors[tableName]
+					if testCase.metricsGrouping == singleTable {
+						createdIngestor = plugin.metricIngestors[tableName]
+					}
+					records := mockIngestor.records[0] // the first element
+					require.NotNil(t, createdIngestor)
+					require.JSONEq(t, jsonValue, records)
+				}
+			}
```

Comment:
This saves an indentation level and makes code easier to read
```suggestion
				if testCase.expectedWriteError != "" {
					require.EqualError(t, errorInWrite, testCase.expectedWriteError)
					continue
				}
				require.NoError(t, errorInWrite)
				createdIngestor := plugin.metricIngestors[tableName]
				if testCase.metricsGrouping == singleTable {
					createdIngestor = plugin.metricIngestors[tableName]
				}
				records := mockIngestor.records[0] // the first element
				require.NotNil(t, createdIngestor)
				require.JSONEq(t, jsonValue, records)
			}
```